R/compile_report.R

```diff
@@ -67,10 +80,68 @@ compile_report <- function(file, quiet = FALSE, factory = getwd(),
   files_before <- unique(sub("~$", "", files_before))
 
 
+  ## handle optional parameters passed through ...:
+
+  ## we extract `params`, display
+  ## some information to the console on parameters used, and generate some text
+  ## that will be used to name the output folder
+  
+  dots <- list(...)
+  has_params <- FALSE
+  if ("params" %in% names(dots)) {
+    ##browser()
+    params <- dots$params
+    
+    ## remove unnamed parameters
+    named <- names(params) != ""
+    params <- params[named]
+
+    if (length(params) > 0) {
+      has_params <- TRUE
+
+      ## convert the parameter values to txt and abbreviate them for display to
+      ## console and for naming the output
+      turn_to_character <- function(x, sep = "_") {
+        out <- as.character(unlist(x))
+        out <- paste(out, collapse = sep)
+        out
+      }
+
+      params_as_txt_console <- lapply(params, turn_to_character, sep = " ")
+      params_as_txt <- lapply(params, turn_to_character)
+      
+      long_values <- nchar(params_as_txt) > 8
+      params_as_txt_console <- lapply(params_as_txt_console, substr, 1, 8)
+      params_as_txt <- lapply(params_as_txt, substr, 1, 8)
+      params_as_txt <- sub("-$", "", params_as_txt)
```

Comment:
what does this do?

Reply:
For all parameters converted to text, we keep only the first 8 characters (that's `substr`); `lapply` applies this in turn to each element of its first argument. The `sub("-$", "", params_as_txt)` makes sure we don't end on a separator (`-` is used to separate values of `params`)

Reply:
👍 thanks